src/pages/learn/[id].tsx

```diff
@@ -67,14 +65,21 @@ function LearnDetail({ videoData, highlightData }: { videoData: VideoData; highl
   };
 
   const handleRightClick = (e: React.MouseEvent, id: number) => {
-    if (isNewMemo) {
-      setClickedScriptId(-1);
-      return setIsConfirmOpen(true);
+    if (highlightData) {
+      const index = highlightData.findIndex((item) => item.highlightId === id);
+      if (index !== undefined && index !== -1) {
+        setHasMemo(Object.keys(highlightData[index].memo).length > 0);
+        setPoints(controlPointX(e));
+        setClickedHighlightId(() => Number((e.target as HTMLDivElement).id));
+        setKeyword((e.target as HTMLDivElement).innerText);
+      }
```

Comment:
```suggestion
      if (index !== undefined && index !== -1) {
        const target = e.target as HTMLDivElement;
        setHasMemo(Object.keys(highlightData[index].memo).length > 0);
        setPoints(controlPointX(e));
        setClickedHighlightId(() => Number(target.id));
        setKeyword(target.innerText);
      }
```

e.target as HTMLDivElement 부분이 중복되어 나타나고 있습니다.
if문 안에 target 변수를 하나 만들어 사용하는 것을 추천합니다 !!